pkg/server/logger.go

```diff
@@ -0,0 +1,46 @@
+package server
+
+import (
+	"net/http"
+
+	"go.uber.org/zap"
+
+	requestcontext "github.com/cmsgov/easi-app/pkg/context"
+)
+
+const traceField string = "traceID"
+
+func loggerMiddleware(logger *zap.Logger, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		traceID, ok := requestcontext.Trace(ctx)
+		if ok {
+			logger = logger.With(zap.String(traceField, traceID.String()))
+		} else {
+			logger.Error("Failed to get trace ID from context")
+		}
+		ctx = requestcontext.WithLogger(ctx, logger)
+
+		fields := []zap.Field{
```

Comment:
This "sortof" has two responsibilities. One to create the logger and other to log the request. I'm open to splitting them out, but it seems close enough and small enough that it would just lead to boilerplate bloat.